sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/SyncToken.java

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.appconfiguration.implementation;
+
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+
+/**
+ * SyncToken class requires id, value and sequence number
+ */
+public final class SyncToken {
+    private String id, value;
+    private long sequenceNumber;
+    private String syncTokenString;
+
+    /**
+     * Create an instance of SyncToken class
+     *
+     * @param id ID of token
+     * @param value Value of Token
+     * @param sequenceNumber Token sequence number (version). Higher value means newer version of the same token.
+     */
+    private SyncToken(String id, String value, long sequenceNumber, String syncTokenString) {
```

Comment:
Why is this constructor private? It looks like you are jumping through hoops in your tests by creating a sync token string and parsing it when you could as easily do `new SyncToken(id, value, sequenceNumber)`. This is an implementation class, we don't expect people to new this up.

Reply:
No other place uses this constructor but only the class itself. Is that making this class singleton would be more approachable?

Reply:
I tried to make this a factory pattern to generate a new sync token.